skimage/segmentation/tests/test_expand_labels.py

```diff
@@ -0,0 +1,189 @@
+)
+
+SAMPLE2D_EXPANDED_3 = np.array(
+      [[1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0],
+       [1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0],
+       [1, 1, 1, 1, 1, 1, 1, 0, 0, 2, 0],
+       [1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 1, 0, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 0, 2, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 0, 0, 2, 2, 2, 2],
+       [0, 0, 1, 0, 0, 0, 0, 2, 2, 2, 2],
+       [0, 0, 0, 0, 0, 0, 0, 0, 0, 2, 0]]
+       )
+
+# non-integer expansion
+SAMPLE2D_EXPANDED_1_5 = np.array(
+      [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+       [1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+       [1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+       [1, 1, 1, 1, 1, 0, 0, 0, 2, 2, 2],
+       [1, 1, 1, 1, 0, 0, 0, 0, 2, 2, 2],
+       [0, 1, 1, 1, 0, 0, 0, 0, 2, 2, 2],
+       [0, 0, 0, 0, 0, 0, 0, 0, 2, 2, 2],
+       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+       [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]])
+
+
+EDGECASE2D =  np.array(
+    [[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+     [0, 0, 0, 0, 0, 2, 2, 0, 0, 0, 0],
+     [0, 0, 1, 1, 0, 2, 2, 0, 0, 0, 0],
+     [0, 1, 1, 1, 0, 2, 0, 0, 0, 0, 0],
+     [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0]]
+)
+
+EDGECASE2D_EXPANDED_4 = array(
+      [[1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 0],
+       [1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2],
+       [1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 0],
+       [1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 0]])
+
+SAMPLE3D = np.array(
+      [[[0, 0, 0, 0],
+        [0, 3, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]],
+
+       [[0, 0, 0, 0],
+        [0, 3, 3, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0]],
+
+       [[0, 0, 0, 0],
+        [0, 3, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 5, 0]],
+
+       [[0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 0, 0],
+        [0, 0, 5, 0]]])
+
+SAMPLE3D_EXPANDED_2 =np.array(
+      [[[3, 3, 3, 3],
+        [3, 3, 3, 3],
+        [3, 3, 3, 3],
+        [0, 3, 5, 0]],
+
+       [[3, 3, 3, 3],
+        [3, 3, 3, 3],
+        [3, 3, 3, 3],
+        [0, 5, 5, 5]],
+
+       [[3, 3, 3, 3],
+        [3, 3, 3, 3],
+        [3, 3, 5, 5],
+        [5, 5, 5, 5]],
+
+       [[3, 3, 3, 0],
+        [3, 3, 3, 0],
+        [3, 3, 5, 5],
+        [5, 5, 5, 5]]])
+
+SAMPLE_EDGECASE_BEHAVIOUR = np.array([[0, 1, 0, 0], [2, 0, 0, 0], [0, 3, 0, 0]])
+
+@testing.parametrize(
+    "input_array, expected_output, expand_distance", 
+    [
+    (SAMPLE1D, SAMPLE1D_EXPANDED_3, 3),
+    (SAMPLE2D, SAMPLE2D_EXPANDED_3, 3),
+    (SAMPLE2D, SAMPLE2D_EXPANDED_1_5, 1.5),
+    (EDGECASE1D, EDGECASE1D_EXPANDED_3, 3),
+    (EDGECASE2D, EDGECASE2D_EXPANDED_4, 4),
+    (SAMPLE3D, SAMPLE3D_EXPANDED_2, 2)
+    ]
+
+)
+def test_expand_labels(input_array, expected_output, expand_distance):
+    expanded = expand_labels(input_array, expand_distance)
+    assert_array_equal(expanded, expected_output)
+
+
+@testing.parametrize(
+    'ndim, distance', list(product([2, 3], range(6))),
+)
```

Comment:
I think that you can chain the decorators and avoid the use/import of `product`:
```suggestion
@testing.parametrize('ndim', [2, 3])
@testing.parametrize('distance', range(6))
```